include/zephyr/arch/common/pm_s2ram.h

```diff
@@ -65,6 +65,11 @@ int arch_pm_s2ram_suspend(pm_s2ram_system_off_fn_t system_off);
  *
  * Default implementation is setting a magic word in RAM. CONFIG_PM_S2RAM_CUSTOM_MARKING
  * allows custom implementation.
+ * The following requirements must be fulfilled:
+ * - the function cannot use stack (asm function or function with 'naked' attribute)
+ * - the content of the R1 and R4 registers must remain unchanged
+ * - returning from the function should be performed with the `bx lr` instruction
```

Comment:
This feels out of place: this header is supposed to be arch-agnostic, but these restrictions are ARM-specific.

Also, as much as I understand the restrictions on `pm_s2ram_mark_check_and_clear`, I don't see why they should apply `pm_s2ram_mark_set` (since it runs on the stack of a regular thread, post OS boot, etc).

Reply:
exactly my thought. this is simply wrong.